subset-routing-test/src/main/java/com/hazelcast/jet/tests/client/subset/routing/SubsetRoutingTest.java

```diff
@@ -0,0 +1,202 @@
+
+        HazelcastClientInstanceImpl clientInstance = getHazelcastClientInstanceImpl(client);
+
+        long validationCount = 0;
+
+        while (System.currentTimeMillis() - begin < durationInMillis) {
+            assertEffectiveMembersAndActiveConnections(clientInstance);
+            assertRequestToCluster(clientInstance);
+
+            validationCount++;
+            if (validationCount % logVerificationCountThreshold == 0) {
+                logger.info(logPrefix + "Validations count: " + validationCount);
+            }
+            sleepMinutes(sleepBeforeValidationInMinutes);
+
+        }
+        logger.info(logPrefix + "Final validations count: " + validationCount);
+    }
+
+    private void assertRequestToCluster(HazelcastClientInstanceImpl clientInstance) {
+        AtomicReference<Throwable> lastException = new AtomicReference<>();
+
+        ConditionVerifierWithTimeout verifier = new ConditionVerifierWithTimeout(
+                Duration.ofMinutes(maxWaitForExpectedConnectionInMinutes),
+                Duration.ofSeconds(sleepBetweenConnectionAssertionAttemptsInSeconds));
+
+        verifier
+                //This method throws OperationTimeoutException for unavailable cluster
+                .condition(() -> clientInstance.getMap(this.getClass().getSimpleName()).isEmpty())
+                .onException(lastException::set)
+                .onTimeout(() -> {
+                    if (lastException.get() != null) {
+                        throw new IllegalStateException("Timed out waiting for cluster active connection",
+                                lastException.get());
+                    }
+                    throw new IllegalStateException("Timed out waiting for cluster active connection");
+                })
+                .execute();
+
+    }
+
+    @Override
+    protected List<ClusterType> runOnClusters() {
+        return List.of(DYNAMIC, STABLE);
+    }
+
+    private void assertEffectiveMembersAndActiveConnections(HazelcastClientInstanceImpl clientInstance) {
+        Duration maxLoopDuration = Duration.ofMinutes(maxWaitForExpectedConnectionInMinutes);
+        Duration sleepBetweenAttempts = Duration.ofSeconds(sleepBetweenConnectionAssertionAttemptsInSeconds);
+
+        ConditionVerifierWithTimeout verifier = new ConditionVerifierWithTimeout(maxLoopDuration, sleepBetweenAttempts);
+
+        verifier.condition(() -> {
+                    Collection<Member> effectiveMemberList = getEffectiveMemberList(clientInstance);
+                    int activeConnections = getActiveConnections(clientInstance);
+                    return effectiveMemberList.size() == 1 && activeConnections == 1;
+                })
+                .onConditionPass(() -> {
+                    UUID newConnectedMemberUuid = getEffectiveMemberList(clientInstance)
+                            .iterator().next().getUuid();
+                    if (!newConnectedMemberUuid.equals(connectedMemberUuid)) {
```

Comment:
And is this actually even ok for stable cluster? Shouldn't be the client connect to one member for the whole test?

Reply:
Fixed. For stable in theory client should be connected to the same member during a test :thinking:  but I'm not sure if I really have to add this kind of assertion :thinking:  If something will happen with connection, or if one of member will have OOM then moving to another member will be expected then. WDYT?

Reply:
> If something will happen with connection, or if one of member will have OOM then moving to another member will be expected the

it could be expected for behavior, but is it expected for the tests on stable cluster?